test/e2e/clusterctl_upgrade_test.go

```diff
@@ -209,7 +209,7 @@ var _ = Describe("When testing clusterctl upgrades (v1.4=>current)", func() {
 			SkipCleanup:               skipCleanup,
 			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.4.0/clusterctl-{OS}-{ARCH}",
 			InitWithProvidersContract: "v1beta1",
-			InitWithKubernetesVersion: "v1.26.0",
+			InitWithKubernetesVersion: "v1.27.0",
```

Comment:
Technically v1.4 will only support running on v1.27 after the first v1.4.x release which supports v1.27 but let's just run the full tests on this PR and then bump on here already.

tl;dr let's run e2e-full on this PR and then this change is fine